homeassistant/components/rituals_perfume_genie/config_flow.py

```diff
@@ -0,0 +1,61 @@
+"""Config flow for Rituals Perfume Genie integration."""
+import logging
+
+from aiohttp import ClientResponseError
+from pyrituals import Account, AuthenticationException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import ACCOUNT_HASH, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Rituals Perfume Genie."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+        errors = {}
+
+        await self.async_set_unique_id(user_input[CONF_EMAIL])
+        self._abort_if_unique_id_configured()
+
+        try:
+            session = async_get_clientsession(self.hass)
+            account = Account(
+                user_input[CONF_EMAIL], user_input[CONF_PASSWORD], session
+            )
```

Comment:
Can these throw an exception?  If not, please move them outside of the try block

Reply:
These should not throw an exception. I moved them outside the try block. Commit: https://github.com/home-assistant/core/pull/46218/commits/5765f023513061bee67facdab693a5d22876011a